test/functional/wallet_backwards_compatibility.py

```diff
@@ -158,34 +173,46 @@ def run_test(self):
 
         # Unload wallets and copy to older nodes:
         node_master_wallets_dir = node_master.wallets_path
-        node_v19_wallets_dir = node_v19.wallets_path
-        node_v17_wallets_dir = node_v17.wallets_path
-        node_v16_wallets_dir = node_v16.chain_path
         node_master.unloadwallet("w1")
         node_master.unloadwallet("w2")
         node_master.unloadwallet("w3")
 
         for node in legacy_nodes:
             # Copy wallets to previous version
             for wallet in os.listdir(node_master_wallets_dir):
-                shutil.copytree(
-                    os.path.join(node_master_wallets_dir, wallet),
-                    os.path.join(self.nodes_wallet_dir(node), wallet)
-                )
-
-        if not self.options.descriptors:
-            # Descriptor wallets break compatibility, only run this test for legacy wallet
-            # Load modern wallet with older nodes
-            for node in legacy_nodes:
-                for wallet_name in ["w1", "w2", "w3"]:
-                    if node.version < 170000:
-                        # loadwallet was introduced in v0.17.0
-                        continue
-                    if node.version < 180000 and wallet_name == "w3":
-                        # Blank wallets were introduced in v0.18.0. We test the loading error below.
-                        continue
-                    node.loadwallet(wallet_name)
-                    wallet = node.get_wallet_rpc(wallet_name)
+                if node.version == 160300:
```

Comment:
v0.16 is different in many regards, does it make sense to have a separate test to check incompatibility?

Reply:
Perhaps, but I think we should test it in this test as it's important to have the upgrade and downgrade tests for the older wallet version.

Reply:
agree, you can resolve this

Reply:
No, this is more correct. This behavior doesn't apply to all older versions, only to 0.16.x. Versions prior to this kept the wallet files in the main datadir. 0.16 added the wallet directory but the wallets are still per file. 0.17 moved wallets into the current one directory per wallet structure.